ts/features/itwallet/lifecycle/saga/checkWalletInstanceStateSaga.ts

```diff
@@ -0,0 +1,50 @@
+import { Errors } from "@pagopa/io-react-native-wallet";
+import { deleteKey } from "@pagopa/io-react-native-crypto";
+import { call, put, select } from "typed-redux-saga/macro";
+import * as O from "fp-ts/lib/Option";
+import { getAttestation } from "../../common/utils/itwAttestationUtils";
+import { ensureIntegrityServiceIsReady } from "../../common/utils/itwIntegrityUtils";
+import { itwIntegrityKeyTagSelector } from "../../issuance/store/selectors";
+import { ReduxSagaEffect } from "../../../../types/utils";
+import { itwLifecycleIsOperationalOrValid } from "../store/selectors";
+import { itwLifecycleWalletReset } from "../store/actions";
+import { walletRemoveCardsByType } from "../../../newWallet/store/actions/cards";
+
+export function* handleWalletInstanceReset(integrityKeyTag: string) {
+  yield* put(itwLifecycleWalletReset());
+  yield* put(walletRemoveCardsByType("itw"));
+  yield* call(deleteKey, integrityKeyTag);
+}
```

Comment:
Should we also purge our persisted storage with the credentials?

Reply:
I think we can let Redux Persist handle it. When the credentials reducer is brought back to its initial empty state with `itwLifecycleWalletReset`, the persisted value is overridden with the empty state. Unless I am missing something else.

Reply:
Makes sense since the `setItem` calls the native implementation which allows data to be overwritten. Thanks.